include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -47,13 +47,61 @@ namespace internal
       return a;
     }
 
-    template <typename Number>
-    Number
-    get_first_array_element(const VectorizedArray<Number> a)
+    float
+    get_first_array_element(const VectorizedArray<float, 1> a)
```

Comment:
I assume this was one of the places you mentioned where the template specialization did not get picked up correctly by the compiler, so you had to specify all widths separately? I would have liked to have
```
template <int width>
float get_first_array_element(const VectorizedArray<float, width> a)
{
  return a[0];
}
template <int width>
double get_first_array_element(const VectorizedArray<double, width> a)
{
  return a[0];
}
```
or even
```
template <typename Number, int width>
double get_first_array_element(const VectorizedArray<Number, width> a)
{
  return a[0];
}
```